python/ray/rllib/agents/ddpg/ddpg_policy_graph.py

```diff
@@ -238,38 +76,47 @@ def __init__(self, observation_space, action_space, config):
                     action_space))
 
         self.config = config
-        self.cur_epsilon = 1.0
+        self.cur_noise_scale = 1.0
+        self.cur_pure_exploration_phase = False
         self.dim_actions = action_space.shape[0]
         self.low_action = action_space.low
         self.high_action = action_space.high
 
         # create global step for counting the number of update operations
         self.global_step = tf.train.get_or_create_global_step()
 
+        # use separate optimizers for actor & critic
+        self._actor_optimizer = tf.train.AdamOptimizer(
+            learning_rate=self.config["actor_lr"])
+        self._critic_optimizer = tf.train.AdamOptimizer(
+            learning_rate=self.config["critic_lr"])
+
         # Action inputs
         self.stochastic = tf.placeholder(tf.bool, (), name="stochastic")
-        self.eps = tf.placeholder(tf.float32, (), name="eps")
-        self.cur_observations = tf.placeholder(
-            tf.float32,
-            shape=(None, ) + observation_space.shape,
-            name="cur_obs")
+        self.noise_scale = tf.placeholder(tf.float32, (), name="noise_scale")
+        self.pure_exploration_phase = tf.placeholder(
+            tf.bool, (), name="pure_exploration_phase")
+        self.cur_observations = tf.placeholder(tf.float32,
+                                               shape=(None, ) +
+                                               observation_space.shape,
+                                               name="cur_obs")
 
-        # Actor: P (policy) network
         with tf.variable_scope(POLICY_SCOPE) as scope:
-            p_values, self.p_model = self._build_p_network(
+            policy_out, self.policy_model = self._build_policy_network(
                 self.cur_observations, observation_space, action_space)
-            self.p_func_vars = _scope_vars(scope.name)
+            self.policy_vars = _scope_vars(scope.name)
 
         # Noise vars for P network except for layer normalization vars
         if self.config["parameter_noise"]:
             self._build_parameter_noise([
-                var for var in self.p_func_vars if "LayerNorm" not in var.name
+                var for var in self.policy_vars if "LayerNorm" not in var.name
             ])
 
         # Action outputs
         with tf.variable_scope(ACTION_SCOPE):
-            self.output_actions = self._build_action_network(
-                p_values, self.stochastic, self.eps)
+            self.output_actions = self._build_exploration_noise(
```

Comment:
Is the function name indicative? As it returns an action instead of a noise alone, I don't think it precisely describes its functionality.